pkgs/applications/window-managers/hyprland/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, libdrm
+, libinput
+, libxcb
+, libxkbcommon
+, mesa
+, pango
+, wayland
+, wayland-protocols
+, wlroots
+, xcbutilwm
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hyprland";
+  version = "0.5.0beta";
+
+  src = fetchFromGitHub {
+    owner = "vaxerski";
+    repo = pname;
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-4dBp2Qlu4RsW5OicYEjl8L/CjJqnZlESS9Mksww44IM=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    wayland
+  ];
+
+  buildInputs = [
+    libdrm
+    libinput
+    libxcb
+    libxkbcommon
+    mesa
+    pango
+    wayland-protocols
+    wlroots
+    xcbutilwm
+  ];
+
+  postPatch = ''
+    make protocols
+  '';
+
+  postBuild = ''
+    pushd ../hyprctl
+    $CXX -std=c++20 -w ./main.cpp -o ./hyprctl
+    popd
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m755 ./Hyprland $out/bin
```

Comment:
You are sure this should be upper cased?

Reply:
Yes, the built binary is uppercased.